_functions_tomcat.sh

```diff
@@ -185,39 +185,47 @@ do_configure_tomcat_datasources() {
       find_instance_file DB_SERVER_PATCH "${ETC_DIR}/${DEPLOYMENT_APPSRV_TYPE}${DEPLOYMENT_APPSRV_VERSION:0:1}" "server-mysql.xml.patch" "${DB_SERVER_PATCH_PRODUCT_NAME}"
 
       # Deploy the Mysql driver
-      if [ ! -f ${DEPLOYMENT_DIR}/lib/mysql-connector*.jar ]; then
-        MYSQL_JAR_URL="http://repository.exoplatform.org/public/mysql/mysql-connector-java/${DEPLOYMENT_MYSQL_DRIVER_VERSION}/mysql-connector-java-${DEPLOYMENT_MYSQL_DRIVER_VERSION}.jar"
-        if [ ! -e ${DL_DIR}/mysql-connector-java/${DEPLOYMENT_MYSQL_DRIVER_VERSION}/`basename ${MYSQL_JAR_URL}` ]; then
-          if ${ADT_OFFLINE}; then
-            echo_error "ADT is offine and the MySQL JDBC Driver isn't available locally"
-            exit 1
-          else
-            mkdir -p ${DL_DIR}/mysql-connector-java/${DEPLOYMENT_MYSQL_DRIVER_VERSION}/
-            echo_info "Downloading MySQL JDBC driver from ${MYSQL_JAR_URL} ..."
-            set +e
-            curl --fail --show-error --location-trusted ${MYSQL_JAR_URL} > ${DL_DIR}/mysql-connector-java/${DEPLOYMENT_MYSQL_DRIVER_VERSION}/`basename ${MYSQL_JAR_URL}`
-            if [ "$?" -ne "0" ]; then
-              echo_error "Cannot download ${MYSQL_JAR_URL}"
-              rm -f "${DL_DIR}/mysql-connector-java/${DEPLOYMENT_MYSQL_DRIVER_VERSION}/"`basename ${MYSQL_JAR_URL}` # Remove potential corrupted file
+      if ${DEPLOYMENT_FORCE_JDBC_DRIVER_ADDON}; then 
+        local addon="exo-jdbc-driver-mysql:${DEPLOYMENT_MYSQL_ADDON_VERSION}"
+        echo_info "Using ${addon} addon as jdbc driver"
+        env_var "DEPLOYMENT_ADDONS" "${DEPLOYMENT_ADDONS},${addon}"
+      else 
```

Comment:
for better readability, you should move the manual driver installation code to a dedicated function do_install_mysql_driver like for other drivers. This will also allow to mutualize the mysql manual installation code between tomcat and jboss use case (like other driver types)

Reply:
done